lms/views/api/canvas/authorize.py

```diff
@@ -20,16 +20,22 @@
 
 @view_defaults(request_method="GET", route_name="canvas_oauth_callback")
 class CanvasAPIAuthorizeViews:
-
-    #: The Canvas API scopes that we request.
-    scopes = (
-        "url:GET|/api/v1/courses/:course_id/files",
-        "url:GET|/api/v1/files/:id/public_url",
-    )
-
     def __init__(self, request):
         self.request = request
 
+        #: The Canvas API scopes that we request.
+        self.scopes = (
+            "url:GET|/api/v1/courses/:course_id/files",
+            "url:GET|/api/v1/files/:id/public_url",
+        )
+
+        if request.feature("section_groups"):
+            self.scopes = self.scopes + (
+                "url:GET|/api/v1/courses/:id",
+                "url:GET|/api/v1/courses/:course_id/sections",
+                "url:GET|/api/v1/courses/:course_id/users/:id",
+            )
```

Comment:
Once the feature flag is removed it'll just be one hard-coded list of scopes that we always request. If you have a developer key / if we're talking to the Canvas API for you at all, then we require both the Files and Sections scopes.

For now, though, we don't want to request the sections scopes when the feature flag is off.

Reply:
Urgh, just realised that if there's a preexisting application instance that does have a developer key, then it needs to request the Canvas files scopes but not the sections ones (even if the feature flag is on). So there's a little more work to do here